src/rdkafka_conf.c

```diff
@@ -1004,15 +1009,69 @@ static const struct rd_kafka_property rd_kafka_properties[] = {
           0, 1, 0,
           _UNSUPPORTED_OAUTHBEARER
         },
-        { _RK_GLOBAL, "oauthbearer_token_refresh_cb", _RK_C_PTR,
-          _RK(sasl.oauthbearer_token_refresh_cb),
+        { _RK_GLOBAL, "sasl.oauthbearer.token_refresh_cb", _RK_C_PTR,
```

Comment:
Must not change the name of existing properties.

Callback properties use snake_case while the ordinary text-based properties use dot-case.